:memo: Updating troubleshooting for git push out of disk issue (#2057)
platformsh/platformsh-docs: docs/src/development/troubleshoot.md

Or you can declare writable [mounts](../configuration/app/storage.md#mounts), which are writable even during and after deploy.
They can be used for your data: file uploads, logs, and temporary files.

## Git push fails due to lack of disk space

If you see the following message when attempting to do a `git push`:
`There is not enough free space to complete the push`

It usually indicates that large files are present in the repository (where they shouldn't be).

> Make sure that the paths for media files, dependencies, or databases, ... are set to be ignored in the `.gitignore` file.

Reviewer: ```suggestion
Make sure that the paths for files like media files, dependencies, and databases are set to be ignored in your `.gitignore` file.
```

The ellipses is essentially like a less clear version of `etc.`, which [Google recommends against](https://developers.google.com/style/word-list?hl=en#etc).

> If large files are already in the repository, the open-source [bfg-repo-cleaner](https://rtyley.github.io/bfg-repo-cleaner/) can assist in cleaning up the repository by purging older commits, removing unnecessary files, ...

Reviewer: ```suggestion
If large files are already in the repository, the open-source tool [bfg-repo-cleaner](https://rtyley.github.io/bfg-repo-cleaner/)
can assist in cleaning up the repository by purging older commits, removing unnecessary files, and more
```

> Create a [support ticket](https://console.platform.sh/-/users/~/tickets/open) to have the disk space temporarily increased.

Reviewer: ```suggestion
To have the disk space temporarily increased, create a [support ticket](https://console.platform.sh/-/users/~/tickets/open).
```

This suggestion is about the general [clause order recommendations]. I'd also consider starting with some more context (like "if this doesn't work" or something).

Also, would it only be for a temporary increase? I thought it wasn't temporary. If it's only temporary, it seems like we should push the other fixes more?

Author: The temporality of the fix depends on the size that is required. As far as I know, some sizes can not be set as a global fix (but I might be wrong). Anyway, repositories that have sizes over 4-8 GB are probable signs of issues in development. 
For e.g [limit in Github is 5GB](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github)

Reviewer: The fact that its a sign of an issue seems to me a sign that we shouldn't be suggesting a temporary fix then. If it's a development issue, then that should be fixed. Otherwise, it's not a temporary fix: it's intentional and should be introduced differently.

Author: You are correct, I removed that part out of the documentation. It usually is a sign of a development issue, and the courtesy that support allows is probably not always a good approach.

Or you can declare writable [mounts](../configuration/app/storage.md#mounts), which are writable even during and after deploy.
They can be used for your data: file uploads, logs, and temporary files.

## Git push fails due to lack of disk space

You might see the following message when attempting to run `git push`:
`There is not enough free space to complete the push`

This usually indicates that large files are present in the repository (where they shouldn't be).
Make sure that the paths for files like media files, dependencies, and databases are set to be ignored in your `.gitignore` file.

If large files are already in the repository, the open-source tool [bfg-repo-cleaner](https://rtyley.github.io/bfg-repo-cleaner/)
can assist in cleaning up the repository by purging older commits, removing unnecessary files, and more.

> If none of the propositions above worked, create a [support ticket](https://console.platform.sh/-/users/~/tickets/open).

Reviewer: ```suggestion
If none of these suggestions work, create a [support ticket](https://console.platform.sh/-/users/~/tickets/open).
```

I'd try to stay away from "propositions" cause it's not so common of a word and "above" for [accessibility reasons](https://developers.google.com/style/accessibility?hl=en) (avoiding "directional" language).

Otherwise, it all looks good to me.
